hera_cal/smooth_cal.py

```diff
@@ -350,9 +351,102 @@ def rephase_to_refant(gains, refant, flags=None, propagate_refant_flags=False):
+    if len(lst_blacklists) > 0:
+        # If lat_lon_alt is not specified, try to infer it from the telescope name, which calfits files generally carry around
+        if lat_lon_alt_degrees is None:
+            if telescope_name.upper() == 'HERA':
+                lat_lon_alt_degrees = np.array(pyuvdata.utils.LatLonAlt_from_XYZ(utils.HERA_TELESCOPE_LOCATION))
+                lat_lon_alt_degrees *= [180 / np.pi, 180 / np.pi, 1]
+            else:
+                raise NotImplementedError(f'No known position for telescope {telescope_name}. lat_lon_alt_degrees must be specified.')
+
+        # calculate LST grid in hours from time grid and lat_lon_alt
+        lst_grid = pyuvdata.utils.get_lst_for_time(time_grid, *lat_lon_alt_degrees) * 12 / np.pi
+
+        # add blacklisted times from lst_blacklists
+        for bounds in lst_blacklists:
+            assert isinstance(bounds, Iterable) and len(bounds) == 2, 'lst_blacklists must be list of pairs of bounds'
+            if bounds[0] < bounds[1]:
+                time_blacklist_array[(lst_grid >= bounds[0]) & (lst_grid <= bounds[1])] = True
+            else:  # the bounds span the 24 hours --> 0 hours branch cut
+                time_blacklist_array[(lst_grid >= bounds[0]) | (lst_grid <= bounds[1])] = True
+
+    return time_blacklist_array
+
+
+def build_freq_blacklist(freqs, freq_blacklists=[], chan_blacklists=[]):
+    '''Converts pairs of bounds on blacklisted frequencies/channels into a boolean array of blacklisted freqs.
+
+    Arguments:
+        freqs: numpy array of frequencies
+        freq_blacklists: list of pairs of frequencies in the same units bounding (inclusively) the spectral regions that
+            are to be marked as True in the freq_blacklist_array
+        chan_blacklists: list of pairs of channel numbers bounding (inclusively) spectral regions that are to be marked 
+            as True in the freq_blacklist_array.
+
+    Returns:
+        freq_blacklist_array: boolean array with the same shape as freqs with blacklisted frequencies set to True'''
+
+    freq_blacklist_array = np.zeros(len(freqs), dtype=bool)
+
+    # Calculate blacklisted frequencies
+    if len(freq_blacklists) > 0:
+        for bounds in freq_blacklists:
+            assert isinstance(bounds, Iterable) and len(bounds) == 2, 'freq_blacklists must be list of pairs of bounds'
+            assert bounds[0] <= bounds[1], 'freq_blacklists bounds must be in ascending order'
+            freq_blacklist_array[(freqs >= bounds[0]) & (freqs <= bounds[1])] = True
+
+    # Calculate blacklisted channels
+    if len(chan_blacklists) > 0:
+        for bounds in chan_blacklists:
+            assert isinstance(bounds, Iterable) and len(bounds) == 2, 'chan_blacklists must be list of pairs of bounds'
+            assert bounds[0] <= bounds[1], 'chan_blacklists bounds must be in ascending order'
+            freq_blacklist_array[(np.arange(len(freqs)) >= bounds[0]) & (np.arange(len(freqs)) <= bounds[1])] = True
+
```

Comment:
Again the chronological order can be determined inside the code itself. This will alleviate user effort and it is straightforward to include.